Add romanesco clone to install docs

mgrauer opened the pull request:
Fixes #169.

ping @aashish24

danlamanna commented:
Can we hold on this? I would rather add git-checkout installation instructions to the romanesco docs and link to those, than put it into Minerva.

If this is okay, I can add them later today.

mgrauer commented:
I agree @danlamanna .  I'll fix this and push new commits.

mgrauer commented:
@danlamanna I made the changes you suggest, but reading those Romanesco docs, I'm confused.

I couldn't find a mention of installing the Romanesco python requirements using pip through the source installation path, and I also couldn't figure out how to enable Romanesco backends (like Spark) through the source installation.

Is there a clearer path we can give for installation?  Perhaps installing Romanesco through pip (specifying [spark]) and then symlinking it into the girder/plugins dir?

jbeezley commented:
Probably this makes most sense.

```
cd /path/to/girder/plugins
git clone https://github.com/Kitware/romanesco.git
cd romanesco
pip install -e .[spark]
```

You can also do
```
pip install git+https://github.com/Kitware/romanesco.git#egg=romanesco[spark]
```

mgrauer commented:
I'm ok with that, thanks for providing this clear path.

@danlamanna I know you wanted to link to the Romanesco docs.  Are you ok with me adding what @jbeezley said, or do you want a chance to update the Romanesco docs?

We can also put in what jbeezley said until the Romanesco docs are clarified at some later date.

danlamanna commented:
Assuming this is no rush - I'll put a section in the Romanesco docs with what @jbeezley said, and then update this.

mgrauer commented:
No rush.

danlamanna commented:
@mgrauer Take a look at Kitware/romanesco#187

mgrauer commented:
This is a good start @danlamanna , but I want to provide more guidance to people who aren't as familiar with Romanesco (me included).

I liked @jbeezley 's explicit instructions

>cd /path/to/girder/plugins
git clone https://github.com/Kitware/romanesco.git
cd romanesco
pip install -e .[spark]

The Romanesco docs do say to turn on Romanesco as a Girder plugin (which you linked to in 9371335), though we will get the Romanesco plugin enabled when Minerva plugin is enabled because of the dependency relationship.  We could link there for further information and to stay current.

Currently we have the following line in our requirements.txt

[-e git+https://github.com/Kitware/romanesco.git#egg=romanesco[spark]](https://github.com/Kitware/minerva/blob/2771a7a3620bb1d955e8460628c439ba92cbe098/requirements.txt#L8)

Should we remove that line if we link to the above explicit instructions?  Note that using the `-e` form as we currently have it also requires [explicitly creating a worker.local.cfg file](https://github.com/Kitware/minerva/blob/2771a7a3620bb1d955e8460628c439ba92cbe098/.travis.yml#L90-L91).  What I don't like about removing this from requirements.txt is that we then only express our requirements in the documentation, and not explicitly for Python, though we do explicitly in our `plugin.json`.


Probably I should have started this PR with the following:

My goals are to inform developers

  * that we depend on Romanesco, as explicitly as possible
  * how to install Romanesco[spark] as a Girder plugin

@danlamanna  do you think you could revise your instructions to accomplish these goals and also include links to the relevant sections in the Rom docs?

Ideally we should follow this up by revising our .travis to install in the same way that we recommend, and to provide docs for how to run Romanesco[spark] for Minerva.

mgrauer commented:
This is what I'll go with

    * install Romanesco in the girder/plugins directory
    * link to "install Rom as a girder plugin"
    * enable the Romanesco[spark] plugin in the girder/plugins/Romanesco/Romanesco[?] dir
    * link to pip -e . [spark] in rom docs

I think I should also update travis to fit with this

mgrauer commented:
This branch will need to be rebased and cleaned up.  I tried to follow the install instructions (which ones?  the ones I was "improving") for a brand new Ubuntu 14.04 VM and ran into many troubles.

The current install instructions work, but I have some questions.

(1)

When I did 

    cd /path/to/girder/plugins 
    git clone https://github.com/Kitware/romanesco.git
    cd romanesco
    pip install -e .[spark]

This did not create the worker.local.cfg, not did it add spark as an enabled plugin to the dist config.  I didn't include the line `pip install -e .[spark]` in the install instructions, though I did run it during the course of my attempted install, so it is possible it is actually needed.

(2)

Doing

    pip install git+https://github.com/Kitware/romanesco.git#egg=romanesco[spark]

didn't install Romanesco as a Girder plugin, though maybe @jbeezley meant to install with git and then also install with pip?

(2.5)

I settled on installing Romanesco with git, and manually copying worker.dist.cfg to worker.local.cfg and enabling the spark plugin in the worker.local.cfg file in Romanesco.  Along with a lot of PATH setting.

(3) I set the SPARK_HOME and PATH for running Girder and running Celery (Romanesco worker).  Are these ENV settings neccessary for both python processes?

(4) I didn't have success installing and building all of the client side code in girder + plugins by only doing `npm install` at the top level of girder.  I needed to do

    cd GIRDER_DIR
    sudo npm install -g npm
    npm install grunt grunt-cli
    npm install

but maybe I'm missing something.

(5) I need to repeat this from end-to-end starting with a fresh Ubuntu 14.04 image.

(6) Is it weird that the Romanesco Girder plugin on Travis doesn't need to set the group/user in the Girder Admin Plugin config of Romanesco?  Or is it expected that a Girder admin user can run a Romanesco job without setting permissions through that config?  I probably need to update the install instructions with the result of this.

(7) Once I get the bugs worked out, I hope to get the .travis.yml working the exact same way, so it reinforces rather than conflicts with these instructions.  That may be wishful thinking.

(8) This is all way too much work to get Minerva installed.  I'm favoring more and more splitting out Romanesco support in Minerva into some separate plugin, see #199.  That sounds even more complicated, but ideally some way of simplifying all this by making Minerva's dependence on Romanesco optional.

mgrauer commented:
I've rebased and cleaned up the install instructions.

(1) The Romanesco[spark] install path for Minerva is documented and in summary

  * `git clone` romanesco into girder/plugins/romanesco
  * `pip install -e .[spark]` in girder/plugins/romanesco
  * `cp worker.dist.cfg worker.local.cfg` in girder/plugins/romanesco/romanesco
  * enable spark plugin in girder/plugins/romanesco/romanesco/worker.local.cfg
  * `python -m romanesco`  in girder/plugins/romanesco, no PATH setting needed

(2) The path to running spark is documented

(3) Other than the install of node, `npm install` is all that is needed to build the Girder, Minerva, Romanesco clients

(4) I'll wait on updating .travis.yml, #191

(5) We can think on whether or how to split out Romanesco functionality in Minerva, #199


@danlamanna PTAL.

danlamanna commented on docs/installation.rst:
```diff
@@ -63,19 +121,97 @@ Install of Minerva as a Girder plugin
+
+
+
+Because Romanesco is a Girder plugin that is a depedency of Minerva, when you enable the Minerva Girder plugin, you will also enable the Romanesco Girder plugin.
```
dependency*

mgrauer commented:
Spelling corrected.

mgrauer commented:
Also fixes #192.

mgrauer commented:
Also fixes #96.
